### PR TITLE
This PR fixes the link to the CHEG (Check How to Export Goods) service

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-1158
See comment.

### What?
This PR fixes the URL to the CHEG (Check How to Export Goods) service.

### Why?
To link Check How to Export Goods service properly.

### CODE REVIEW

File: app/views/measures/_export_tab_check_duties.html.erb [hunk 1]
```diff
@@ -17,11 +17,11 @@
...
       <% if european %>
-        <%= link_to("Check how to export #{commodity_code} to #{country_name} (link opens in new tab)",
-                    "https://www.check-duties-customs-exporting-goods.service.gov.uk/summary?d=#{country_code}&ds=gtp&tab=tree&pc=#{commodity_code}", target: "_blank") %>
+        <%= check_how_to_export_goods_link(commodity_code: commodity_code,
+                                           country_code: country_code,
+                                           country_name: country_name ) %>
       <% else %>
-          <%= link_to('Check how to export commodity goods (link opens in new tab)',
-                    'https://www.check-duties-customs-exporting-goods.service.gov.uk', target: "_blank") %>
+          <%= link_to('Check how to export commodity goods (link opens in new tab)', MeasuresHelper::CHEG_SERVICE_URL, target: "_blank") %>
```
Comment (willfish) [inline app/views/measures/_export_tab_check_duties.html.erb]: Do we want a helper for this link, too?

Comment (alexdesi) [reply app/views/measures/_export_tab_check_duties.html.erb]: not sure, it is just a link with no params,  I think it's ok as it is.

File: app/helpers/measures_helper.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 module MeasuresHelper
+  CHEG_SERVICE_URL = 'https://www.check-duties-customs-exporting-goods.service.gov.uk'.freeze
```
Comment (willfish) [inline app/helpers/measures_helper.rb]: I think we're not supposed to use the CHIEG acronym anymore.

```suggestion
  CHECK_DUTIES_SERVICE_URL = 'https://www.check-duties-customs-exporting-goods.service.gov.uk'.freeze
```

Comment (willfish) [reply app/helpers/measures_helper.rb]: https://www.urbandictionary.com/define.php?term=Chieg

File: app/helpers/measures_helper.rb [hunk 3]
```diff
@@ -21,4 +23,19 @@ def legal_act_regulation_url_link_for(measure)
...
+    link_to("Check how to export #{commodity_code} to #{country_name} (link opens in new tab)",
+            "#{CHEG_SERVICE_URL}/summary?d=#{country_code}&ds=gtp&tab=tree&pc=#{cheg_commodity_code}", target: '_blank', rel: 'noopener')
+  end
+
+  private
+
+  # Right-Strip of the zeros in pairs.
+  # Example: 8514101000 -> 85141010
+  def commodity_code_for_cheg_service(commodity_code)
+    commodity_code.gsub(/(00$|0000$)/, '')
```
Comment (willfish) [inline app/helpers/measures_helper.rb]:
```suggestion
    commodity_code.gsub(/(00|0000)$/, '')
```

File: spec/helpers/measures_helper_spec.rb [hunk 4]
```diff
@@ -44,4 +44,24 @@
       it { expect(helper.legal_act_regulation_url_link_for(measure)).to be_html_safe }
     end
   end
+
+  describe 'check_how_to_export_goods_link' do
+    subject(:link) { check_how_to_export_goods_link(commodity_code: commodity_code, country_code: '', country_name: '') }
```
Comment (willfish) [inline spec/helpers/measures_helper_spec.rb]: Foregoing the helper has weird behaviour with assigns in my experience of writing helper specs

```suggestion
    subject(:link) { helper.check_how_to_export_goods_link(commodity_code: commodity_code, country_code: '', country_name: '') }
```

File: app/helpers/measures_helper.rb (hunk 3 above)
Comment (willfish) [reply app/helpers/measures_helper.rb]: I'd move this to the model

File: app/views/measures/_export_tab_check_duties.html.erb [hunk 5]
```diff
@@ -16,12 +16,9 @@
...
-        <%= link_to("Check how to export #{commodity_code} to #{country_name} (link opens in new tab)",
-                    "https://www.check-duties-customs-exporting-goods.service.gov.uk/summary?d=#{country_code}&ds=gtp&tab=tree&pc=#{commodity_code}", target: "_blank") %>
-      <% else %>
-          <%= link_to('Check how to export commodity goods (link opens in new tab)',
-                    'https://www.check-duties-customs-exporting-goods.service.gov.uk', target: "_blank") %>
-      <% end %>
+      <%= check_how_to_export_goods_link(commodity_code: commodity_code,
+                                         country_code: country_code,
+                                         country_name: country_name,
+                                         eu_member: european ) %>
```
Comment (willfish) [inline app/views/measures/_export_tab_check_duties.html.erb]:
```suggestion
                                         eu_member: eu_member ) %>
```

File: app/views/measures/_export_tab_check_duties.html.erb (hunk 1 above)
Comment (alexdesi) [reply app/views/measures/_export_tab_check_duties.html.erb]: refactored into the helper.

File: app/helpers/measures_helper.rb [hunk 6]
```diff
@@ -21,4 +23,24 @@ def legal_act_regulation_url_link_for(measure)
...
+    else
+      link_to('Check how to export commodity goods (link opens in new tab)', CHECK_DUTIES_SERVICE_URL, target: '_blank', rel: 'noopener')
+    end
+  end
+
+  private
+
+  # Right-Strip of the zeros in pairs.
+  # Example: 8514101000 -> 85141010
+  def commodity_code_for_cheg_service(commodity_code)
```
Comment (willfish) [inline app/helpers/measures_helper.rb]: So the commodity knows about the format of its own commodity code. I think we should move this to the model which is the single source of truth for how the commodity code gets presented. It's also easier to decouple from the link and test in isolation

File: app/helpers/measures_helper.rb (hunk 3 above)
Comment (alexdesi) [reply app/helpers/measures_helper.rb]: Declarable seems to be a good place

Comment (willfish) [reply app/helpers/measures_helper.rb]: Yeah, agree that makes loads of sense

Comment (willfish) [reply app/helpers/measures_helper.rb]: Or goods_nomenclature.rb

File: app/helpers/measures_helper.rb (hunk 6 above)
Comment (alexdesi) [reply app/helpers/measures_helper.rb]: yeah I agree 100%,
I've moved the test too.

File: app/helpers/measures_helper.rb [hunk 7]
```diff
@@ -21,4 +23,14 @@ def legal_act_regulation_url_link_for(measure)
 
     sanitize(link, attributes: %w[href target role rel class title])
   end
+
+  def check_how_to_export_goods_link(commodity_code:, country_code:, country_name:, eu_member:)
+    if eu_member
+      link_to("Check how to export #{commodity_code} to #{country_name} (link opens in new tab)",
+              "#{CHECK_DUTIES_SERVICE_URL}/summary?d=#{country_code}&ds=gtp&tab=tree&pc=#{commodity_code}",
```
Comment (willfish) [inline app/helpers/measures_helper.rb]: I think it makes sense for service config to go in `TradeTariffFrontend.check_duties_service_url`

Comment (willfish) [reply app/helpers/measures_helper.rb]:
```ruby
module TradeTariffFrontend
  def self.check_duties_service_url
    ENV.fetch('CHECK_DUTIES_SERVICE_URL', 'https://www.check-duties-customs-exporting-goods.service.gov.uk')
  end
end
```

**Review by willfish — APPROVED** [review]

**Pull request merged**